receiver/chronyreceiver/factory.go

```diff
@@ -34,13 +34,15 @@ func newMetricsReceiver(
 		return nil, fmt.Errorf("wrong config provided: %w", errInvalidValue)
 	}
 
-	chronyc, err := chrony.New(cfg.Endpoint, cfg.Timeout)
-	if err != nil {
-		return nil, err
-	}
+	s := newScraper(ctx, cfg, set)
 	scraper, err := scraperhelper.NewScraper(
 		metadata.Type.String(),
-		newScraper(ctx, chronyc, cfg, set).scrape,
+		s.scrape,
+		scraperhelper.WithStart(func(ctx context.Context, host component.Host) error {
+			chronyc, err := chrony.New(cfg.Endpoint, cfg.Timeout)
+			s.client = chronyc
+			return err
```

Comment:
Can we move it to s.start?

Reply:
I was thinking if we can add `cfg` to `chronyScraper` struct and add a `start` method. But I'm good with this inline function as well